src/test/java/com/google/sps/servlets/DeleteCardServletTest.java

```diff
@@ -0,0 +1,139 @@
+
+  private HttpServletRequest mockRequest;
+  private HttpServletResponse mockResponse;
+  private StringWriter responseWriter;
+  private DeleteCardServlet servlet;
+  private DatastoreService datastore;
+
+  @Before
+  public void setUp() throws Exception {
+    helper.setUp();
+    servlet = new DeleteCardServlet();
+    mockRequest = mock(HttpServletRequest.class);
+    mockResponse = mock(HttpServletResponse.class);
+
+    // Set up a fake HTTP response
+    responseWriter = new StringWriter();
+    when(mockResponse.getWriter()).thenReturn(new PrintWriter(responseWriter));
+
+    datastore = DatastoreServiceFactory.getDatastoreService();
+  }
+
+  @After
+  public void tearDown() throws Exception {
+    helper.tearDown();
+  }
+
+  @Test
+  public void DeleteCard() throws Exception {
+    // Generate testing card to add into datastore
+    Card card =
+        new Card.Builder()
+            .setImageBlobKey("null")
+            .setRawText("hello")
+            .setTextTranslated("hola")
+            .build();
+
+    // Generate a folder entity to obtain a folder key
+    // which would be used to set as the parent of the card entity
+    Entity folder = new Entity("Folder", "testID");
+    String folderKey = KeyFactory.keyToString(folder.getKey());
+
+    Card cardInDatastore = storeCardInDatastore(card, datastore, folderKey);
+    String cardKey = cardInDatastore.getCardKey();
+
+    when(mockRequest.getParameter("cardKey")).thenReturn(cardKey);
+
+    servlet.doPost(mockRequest, mockResponse);
+    assertEquals(
+        0,
+        datastore
+            .prepare(new Query("Card").setAncestor(folder.getKey()))
+            .countEntities(withLimit(10)));
+  }
+
+  @Test
+  public void DeleteCardAndThereAreOneLeft() throws Exception {
+    // Generate testing cards to add into datastore
+    Card cardA =
+        new Card.Builder()
+            .setImageBlobKey("null")
+            .setRawText("hello")
+            .setTextTranslated("hola")
+            .build();
+    Card cardB =
+        new Card.Builder()
+            .setImageBlobKey("null")
+            .setRawText("hello")
+            .setTextTranslated("xin chào")
+            .build();
+
+    // Generate a folder entity to obtain a folder key
+    // which would be used to set as the parent of the card entity
+    Entity folder = new Entity("Folder", "testID");
+    String folderKey = KeyFactory.keyToString(folder.getKey());
+
+    storeCardInDatastore(cardB, datastore, folderKey);
+    storeCardInDatastore(cardA, datastore, folderKey);
+    String cardAKey = cardA.getCardKey();
+
+    when(mockRequest.getParameter("cardKey")).thenReturn(cardAKey);
+
+    servlet.doPost(mockRequest, mockResponse);
+    assertEquals(
+        1,
+        datastore
+            .prepare(new Query("Card").setAncestor(folder.getKey()))
+            .countEntities(withLimit(10)));
+  }
+
+  public Card storeCardInDatastore(Card card, DatastoreService datastore, String folderKey) {
```

Comment:
I see this method on a bunch of tests — can you find a way to consolidate?

Reply:
done - put in card class.